cairo-proof-parser/src/vec_felt.rs

```diff
@@ -0,0 +1,132 @@
+use serde::{de::Visitor, Deserialize};
+use serde_json::Value;
+use starknet_crypto::Felt;
+use std::{ops::Deref, str::FromStr};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum VecFeltError {
+    #[error(transparent)]
+    NumberParseError(#[from] std::num::ParseIntError),
+    #[error(transparent)]
+    BigIntParseError(#[from] num_bigint::ParseBigIntError),
+    #[error("number out of range")]
+    NumberOutOfRange,
+    #[error(transparent)]
+    Parse(#[from] serde_json::Error),
+}
+
+/// `VecFelt` is a wrapper around a vector of `Arg`.
+///
+/// It provides convenience methods for working with a vector of `Arg` and implements
+/// `Deref` to allow it to be treated like a vector of `Arg`.
+#[derive(Debug, Clone)]
+pub struct VecFelt(Vec<Felt>);
+
+impl VecFelt {
+    /// Creates a new `VecFelt` from a vector of `Arg`.
+    ///
+    /// # Arguments
+    ///
+    /// * `args` - A vector of `Arg`.
+    ///
+    /// # Returns
+    ///
+    /// * `VecFelt` - A new `VecFelt` instance.
+    #[must_use]
+    pub fn new(args: Vec<Felt>) -> Self {
+        Self(args)
+    }
+}
+
+impl IntoIterator for VecFelt {
+    type Item = Felt;
+    type IntoIter = std::vec::IntoIter<Self::Item>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl Deref for VecFelt {
+    type Target = Vec<Felt>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<VecFelt> for Vec<Felt> {
+    fn from(args: VecFelt) -> Self {
+        args.0
+    }
+}
+
+impl From<Vec<Felt>> for VecFelt {
+    fn from(args: Vec<Felt>) -> Self {
+        Self(args)
+    }
+}
+
+impl VecFelt {
+    fn visit_seq_helper(seq: &[Value]) -> Result<Self, VecFeltError> {
```

Comment:
This seems a bit complicated, I suggest creating an enum that would be easily deserializable and adding a converter function to `VecFelt`. The logic will remain the same but the intent would be much clearer